assets/js/payment-methods-demo/payment-methods/stripe/payment-method.js

```diff
@@ -0,0 +1,317 @@
+						) }
+						<CardCvcElement
+							onChange={ onChange }
+							classes={ classes }
+							style={ style }
+						/>
+					</label>
+				</div>
+			</div>
+		</div>
+	);
+};
+
+// @todo add intents?
+
+/**
+ * Stripe Credit Card component
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const CreditCardComponent = ( {
+	paymentStatus,
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit: onCheckoutSubmit,
+	getCustomerPaymentMethods,
+	stripe,
+} ) => {
+	const [ savedPaymentMethodToken, setSavedPaymentMethodToken ] = useState(
+		null
+	);
+	const [ source, setSource ] = useState( null );
+	const customerPaymentMethods = getCustomerPaymentMethods(
+		'cc',
+		PAYMENT_METHOD_NAME
+	);
+	const onStripeError = ( event ) => {
+		const type = event.error.type;
+		const code = event.error.code || '';
+		let message = getErrorMessageForTypeAndCode( type, code );
+		message = message || event.error.message;
+		paymentStatus.setPaymentStatus().error( message );
+		setSource( null );
+
+		// @todo we'll want to do inline invalidation errors for any element
+		// inputs
+	};
+	const updateCardBrand = ( brand ) => {
+		// @todo this updates the card brand logo with non-inline CC forms
+		// (see function in stripe.js)
+	};
+	const onChange = ( paymentEvent ) => {
+		if ( paymentEvent.elementType === 'cardNumber' ) {
+			updateCardBrand( paymentEvent.brand );
+		}
+		if ( paymentEvent.error ) {
+			onStripeError( paymentEvent );
+		} else {
+			setSource( null );
+		}
+	};
+	const renderedCardElement = getStripeServerData().inline_cc_form ? (
+		<CardElement
+			style={ elementStyles }
+			onChange={ onChange }
+			hidePostalCode={ true }
+		/>
+	) : (
+		<CardElements
+			style={ elementStyles }
+			classes={ elementClasses }
+			onChange={ onChange }
+		/>
+	);
+	useEffect( () => {}, [] );
+
+	const createSource = async ( stripeBilling ) => {
+		return await stripe.createSource( stripeBilling );
+	};
+	const onSubmit = () => {
+		const { billingData } = billing;
+		const stripeBilling = {
+			address: {
+				line1: billingData.address_1,
+				line2: billingData.address_2,
+				city: billingData.city,
+				state: billingData.state,
+				postal_code: billingData.postcode,
+				country: billingData.country,
+			},
+		};
+		if ( billingData.phone ) {
+			stripeBilling.phone = billingData.phone;
+		}
+		if ( billingData.email ) {
+			stripeBilling.email = billingData.email;
+		}
+		if ( billingData.first_name || billingData.last_name ) {
+			stripeBilling.name = `${ billingData.first_name } ${ billingData.last_name }`;
+		}
+
+		const response = createSource( stripeBilling );
+		if ( response.error ) {
+			// @todo set payment error
+			// return error object including any validation errors if present?
+		}
+		paymentStatus.setPaymentStatus().success( billingData, {
+			sourceId: response.source.id,
+		} );
+
+		// this should hook into checkout processing? Basically when the checkout
```

Comment:
Am I reading this comment correctly and you want to do 2 actions following this; 1. Process payment, then 2. Update order/set paid?

If so I think /checkout would be a single action with response. Stripe would update the order on the server side after payment, or return an error.

Reply:
> Am I reading this comment correctly and you want to do 2 actions following this; 1. Process payment, then 2. Update order/set paid?

No. What happens in this logic is the `createSource`  method is called to get the stripe _source_ (which is basically the token for completing the payment server side). If the source is successfully created then this `onSubmit` method (which is hooked into the checkout context `onCheckoutProcessing` event, will just return `true`.  

The `setPaymentStatus` is setting the source id (and I have to set the `payment_method` property yet too). 

Then after all the subscribers hooked into the checkout context `onCheckoutProcessing` event are done, checkout hit the server (the new endpoint you are doing) with the package for finishing the checkout processing server side.